File: simple_linear_regression_nidhi.py
# -*- coding: utf-8 -*-
"""
Created on Thu Oct 11 23:30:16 2018

@author: Nidhi
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Salary_Data.csv')
X= dataset.iloc[:,:-1].values
y= dataset.iloc[:,1].values

from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=1/3,random_state=0)

# Fitting Simple Linear Regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting the test set results
y_pred = regressor.predict(X_test)


# Visualization
plt.scatter(X_train,y_train,color='red')
plt.plot(X_train,regressor.predict(X_train), color='blue')
plt.title('Salary Vs Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

File: simple_linear_regression_nidhi.R
dataset = read.csv('Salary_Data.csv')

library(caTools)
set.seed(123)
split=sample.split(dataset$Salary,SplitRatio = 2/3)
training_set = subset(dataset,split == TRUE)
test_set = subset(dataset,split==FALSE)


#Simple Linear Regression
regressor = lm(formula = Salary~YearsExperience, training_set)

#predicting test set results
y_pred = predict(regressor, newdata = test_set)

